File: years/2022/clojure/aoc-2022/test/aoc_2022/day_12_test.clj
(ns aoc-2022.day-12-test
  (:require [aoc-2022.day-12 :refer :all]
            [clojure.test :refer :all]))

(deftest day-12-test
  (testing "sample data"
    (is (= 31 (part1 "../../data/12-sample.txt")))
    (is (= 29 (part2 "../../data/12-sample.txt"))))
  (testing "puzzle data"
    (is (= 330 (part1 "../../data/12.txt")))
    (is (= 321 (part2 "../../data/12.txt")))))

File: years/2024/elixir/test/aoc/day_12_test.exs
defmodule AoC.Day12.Test do
  @moduledoc false

  use ExUnit.Case, async: false

  alias AoC.Day12

  # comment this out to always log to the console
  @moduletag :capture_log

  describe "part 1" do
    test "with sample data" do
      assert Day12.part_1(["../data/12.txt"]) == 1
    end
  end

  describe "part 2" do
    test "with sample data" do
      assert Day12.part_2(["../data/12.txt"]) == 1
    end
  end
end
